src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -258,6 +289,42 @@ auto StreamReader::generic_filter_log_events(
         }
     }
 }
+
+template <typename LogEvent, typename Iterator>
+requires requires(
+                 LogEventWithFilterData<LogEvent> const& event,
+                 clp::ir::epoch_time_ms_t timestamp
+         ) {
+    {
+        event.get_timestamp()
+    } -> std::convertible_to<clp::ir::epoch_time_ms_t>;
+}
+auto StreamReader::generic_get_log_event_index_by_timestamp(
+        Iterator begin,
+        Iterator end,
```

Comment:
Can we pass in the vector directly? e.g. `LogEvents<LogEvent> const& log_events`